src/CMakeLists.txt

```diff
@@ -170,6 +170,10 @@ macro (add_compiler_flag flag)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag}")
 endmacro (add_compiler_flag)
 
+# Disallow undefined symbols in object files. Undefined symbols in shared
+# libraries are still allowed.
```

Comment:
I don't teally see the point of doing this. I thought '-z,defs' will force us to explicitly have to list all dependent libraries that define symbols required by the libraries we link but don't explicitly link -- as it has to be done, in my experience, for static linking, and so this is would be more prudent that way because all requeid libraries would now be listed explicitly on the link line. But based on the above stackowerflo answer that is not the point of this linker flags?